commit 0565ac5ebe9a107393986997adb5d1c2c6063f87 (HEAD -> master)
Author: Isuru Fernando <isuruf@gmail.com>
Date:   Thu Apr 9 18:08:39 2020 +0000

    Remove linking liblldb to libpython
    
    liblldb doubles as an extension module too, but python
    executable is linked to static python. When the extension module
    is loaded both the shared libpython and the static libpython symbols
    are loaded which leads to a crash in OSX.
    
    To fix this, dependence on PYTHON_LIBRARY is removed from liblldb
    and added to its interfaces to make sure the executables like
    lldb works fine.

diff --git a/source/API/CMakeLists.txt b/source/API/CMakeLists.txt
index e0ecf29..35259cf 100644
--- a/source/API/CMakeLists.txt
+++ b/source/API/CMakeLists.txt
@@ -185,6 +185,12 @@ if ( CMAKE_SYSTEM_NAME MATCHES "Windows" )
     target_link_libraries(liblldb PRIVATE ${PYTHON_LIBRARIES})
   endif()
 else()
+  target_link_libraries(liblldb INTERFACE ${PYTHON_LIBRARIES})
+  if ( CMAKE_SYSTEM_NAME MATCHES "Darwin" )
+    target_link_options(liblldb PRIVATE -undefined dynamic_lookup)
+  else()
+    target_link_options(liblldb PRIVATE -Wl,--unresolved-symbols=ignore-all)
+  endif()
   set_target_properties(liblldb
     PROPERTIES
     OUTPUT_NAME lldb
diff --git a/source/Plugins/ScriptInterpreter/Python/CMakeLists.txt b/source/Plugins/ScriptInterpreter/Python/CMakeLists.txt
index 761772f..bd87e6f 100644
--- a/source/Plugins/ScriptInterpreter/Python/CMakeLists.txt
+++ b/source/Plugins/ScriptInterpreter/Python/CMakeLists.txt
@@ -19,7 +19,6 @@ add_lldb_library(lldbPluginScriptInterpreterPython PLUGIN
     lldbHost
     lldbInterpreter
     lldbTarget
-    ${PYTHON_LIBRARIES}
     ${LLDB_LIBEDIT_LIBS}
 
   LINK_COMPONENTS